src/core/activity/plotPanels/interSpikeIntervalHistogramPanel.ts

```diff
@@ -32,34 +32,16 @@ export class InterSpikeIntervalHistogramPanel extends SpikeTimesPanel {
   }
 
   /**
-   * Initialize histogram panel for inter-spike intervals (ISI).
+   * Update data for ISI histogram.
    */
-  init(): void {
-    // console.log('Init histogram panel for inter-spike interval');
-    this.activities = this.graph.project.spikeActivities;
-    this.data = [];
-  }
-
-  /**
-   * Update histogram panel for inter-spike intervals (ISI).
-   *
-   * @remarks
-   * It requires activity data.
-   */
-  update(): void {
-    // console.log('Init histogram panel of spike times')
-    this.activities.forEach((activity: SpikeActivity) => {
-      this.updateInterSpikeIntervalHistogram(activity);
-    });
-    this.layout.xaxis.type = this.state.xaxisType;
-    this.layout.xaxis.title = 'Inter-spike interval [ms]';
-  }
+  updateData(activity: SpikeActivity): void {
+    // console.log('Update data fir ISI histogram.').
```

Comment:
Or could this line even be deleted?
```suggestion
    // console.log('Update data for ISI histogram.').
```